File: LeetCode/541反转字符串Ⅱ.go
func reverseStr(s string, k int) string {
	str := []byte(s)
	for i := 0; i < len(s); i += 2 * k {
		left := i
		right := int(math.Min(float64(left+k-1), float64(len(s)-1)))
		for left < right {
			str[left], str[right] = str[right], str[left]
			left++
			right--
		}
	}
	return string(str)
}

File: LeetCode/541反转字符串Ⅱ.java
class Solution {
    public String reverseStr(String s, int k) {
        char[] chs = s.toCharArray();
        for (int i = 0; i < chs.length; i += 2 * k) {
            int left = i;
            int right = Math.min(left + k - 1, chs.length - 1);
            while (left < right) {
                chs[left] ^= chs[right];
                chs[right] ^= chs[left];
                chs[left] ^= chs[right];
                left++;
                right--;
            }
        }
        return new String(chs);
    }
}